### PR TITLE
chore: Select component refactoring - FilterControl - Iteration 5

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
It replaces the react-select Select component with the Antdesign one in the FilterControl components.

### BEFORE/AFTER

https://user-images.githubusercontent.com/60598000/126142095-30fa08a9-beb2-4098-95f0-753f2fcbd64c.mp4

https://user-images.githubusercontent.com/60598000/126144838-20b68e04-2bb4-4acb-9f44-27e2dc8712cf.mp4

### BEFORE/AFTER

https://user-images.githubusercontent.com/60598000/126142234-93cb7da6-da2e-498b-9118-9e705402bd1f.mp4

https://user-images.githubusercontent.com/60598000/126144743-79a2adcb-859b-4a97-913b-3b8b0e98017b.mp4

### TESTING INSTRUCTIONS
1. Open a chart in Explore
2. Add Filters using the SIMPLE tab options and observe the behavior
3. Add Filters using the CUSTOM SQL options and observe the behavior

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by geido** [comment]
@rosemarie-chiu @jinghua-qa

File: superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterEditPopoverSqlTabContent/index.jsx
Comment (michael-s-molina) [inline superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterEditPopoverSqlTabContent/index.jsx:52]:
```suggestion
const StyledSelect = styled(Select)`
  ${({ theme }) => `
    width: ${theme.gridUnit * 30}px;
    marginRight: ${theme.gridUnit}px;
  `}
`;
```

**Comment by michael-s-molina** [comment]
I think we need to clear the filter value when changing the column.


https://user-images.githubusercontent.com/70410625/133511351-cb519c45-b126-493e-9c56-d24ee0296b3f.mov

**Comment by pkdotson** [comment]
The code lgtm. But in regards to clearing the fields like @michael-s-molina suggested would be nice but to be fair it doesn't currently do that on master so I think that could be done as a follow up as that might require a little more refactoring work?

**Comment by geido** [comment]
> I think we need to clear the filter value when changing the column.
> 
>  Screen.Recording.2021-09-15.at.6.19.33.PM.mov

Just tested on master. As @pkdotson reported this is currently not implemented. I think it makes sense to tackle that in a separate PR as it would be slightly unrelated to the objective of this PR. I'll work on that as soon as we merge this PR.

**Review by michael-s-molina — APPROVED** [review]

**Comment by michael-s-molina** [comment]
> > I think we need to clear the filter value when changing the column.
> 
> @michael-s-molina Just tested on master. As @pkdotson reported this is currently not implemented. I think it makes sense to tackle that in a separate PR as it would be slightly unrelated to the objective of this PR. I'll work on that as soon as we merge this PR.

Ok. Approving with the follow-up in mind.

**Pull request merged**
